freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -98,72 +120,256 @@ def _get_requests_session(cls, retry_options={}):
         the advisory's.
 
-        :param event FlatpakModuleAdvisoryReadyEvent: The event this handler
-            is currently handling.
+        :param images list: A list of image NVRS.
+        :return: a list of ContainerImage instances which can be auto rebuilt.
         :rtype: list
-        :return: List of ContainerImage instances which can be auto rebuilt.
         """
-        errata = Errata()
-        errata_rpm_nvrs = errata.get_cve_affected_rpm_nvrs(event.advisory.errata_id)
+        errata_rpm_nvrs = self.errata.get_cve_affected_rpm_nvrs(
+            self.event.advisory.errata_id
+        )
         lb = LightBlue(
             server_url=conf.lightblue_server_url,
             cert=conf.lightblue_certificate,
             private_key=conf.lightblue_private_key,
             event_id=self.current_db_event_id,
         )
-        images = lb.get_images_by_nvrs(rebuild_images_list, rpm_nvrs=errata_rpm_nvrs)
+        images = lb.get_images_by_nvrs(images, rpm_nvrs=errata_rpm_nvrs)
         return images
+
+    def _parse_compose_source(self, source_value, name_stream_only=True):
+        """
+        Parse source field in a compose and generate a required name_stream_set
+        and an optional name_stream:name_stream_version mapping.
+
+        :param source_value string: The string in a compose source field.
+        :param name_stream_only boolean: True only generate name_stream set,
+            False also generate a mapping between name_stream and name_stream_version.
+        :return: a tuple of name_stream_set and name_stream:name_stream_version mapping.
+        :rtype: tuple
+        """
+        mapping = {}
+        name_stream_set = set()
+        for source in source_value.split():
+            source = source.strip()
+            name_stream = source.rsplit(":", 2)[0]
+            name_stream_set.add(name_stream)
+            if name_stream_only is False:
+                name_stream_version = source.rsplit(":", 1)[0]
+                mapping[name_stream] = name_stream_version
+        return name_stream_set, mapping
+
+    def _outdated_composes(self, original_odcs_compose_ids, module_name_stream_set):
+        """
+        Generate outdated composes.
+
+        :param original_odcs_compose_ids list: Original compose ids.
+        :param module_name_stream_set set: Module name stream set from an advisory.
+        :return: a set of outdated composes.
+        :rtype: set
+        """
+        outdated_composes = set()
+        for compose_id in original_odcs_compose_ids:
+            compose = self.odcs.get_compose(compose_id)
+            source_type = compose.get("source_type")
+            if source_type != PungiSourceType.MODULE:
+                outdated_composes.add(compose_id)
+                continue
+
+            name_stream_set, _ = self._parse_compose_source(compose.get("source", ""))
+
+            # If all the sources are not in advisory, renew them
```

Comment:
No need for comment - the code is clear.

```suggestion
```

Reply:
Ok. I was thinking to emphasize advisory, but module_name_stream_set's meaning can be seen in doc string.